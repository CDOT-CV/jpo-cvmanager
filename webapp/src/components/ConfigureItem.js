import React, { Component } from 'react';

import './css/ConfigureItem.css';

class ConfigureItem extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selected: false,
    }
  }

  toggleSelect = () => {
    let select = !(this.state.selected)
    this.props.updateRsu(this.props.index, select)
    this.setState({'selected': select})
  }

  render() {
    return (
      <div id={this.props.indexList.includes(this.props.index) ? "selectedconfigitemdiv" : "configitemdiv"} 
            onClick={() => this.toggleSelect()}>
        <p id="configitemtext">
          <strong>{this.props.ip}</strong>
        </p>
      </div>
    )
  }
}

export default ConfigureItem;