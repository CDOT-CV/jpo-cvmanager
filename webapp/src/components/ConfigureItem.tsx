import React, { useState } from 'react'

import './css/ConfigureItem.css'

interface ConfigureItemProps {
  ip: string
  index: number
  indexList: number[]
  updateRsu: (index: number, selected: boolean) => void
}

const ConfigureItem = (props: ConfigureItemProps) => {
  const [selected, setSelected] = useState(false)

  const toggleSelect = () => {
    let localSelect = !selected
    props.updateRsu(props.index, localSelect)
    setSelected(localSelect)
  }

  return (
    <div
      id={props.indexList.includes(props.index) ? 'selectedconfigitemdiv' : 'configitemdiv'}
      onClick={() => toggleSelect()}
    >
      <p id="configitemtext">
        <strong>{props.ip}</strong>
      </p>
    </div>
  )
}

export default ConfigureItem
