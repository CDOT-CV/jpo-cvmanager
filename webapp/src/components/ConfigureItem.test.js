import React from "react";
import { render } from "@testing-library/react";
import ConfigureItem from "./ConfigureItem";

it("should take a snapshot", () => {
  const { asFragment } = render(<ConfigureItem indexList={[]} />);

  expect(asFragment(<ConfigureItem indexList={[]} />)).toMatchSnapshot();
});
