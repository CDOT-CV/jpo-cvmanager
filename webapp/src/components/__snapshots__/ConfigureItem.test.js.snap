// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    id="configitemdiv"
  >
    <p
      id="configitemtext"
    >
      <strong />
    </p>
  </div>
</DocumentFragment>
`;
